lib/graphql/fragment_cache/object/object_patch.rb

```diff
@@ -5,9 +5,19 @@ module FragmentCache
     module Object
       # Adds #cache_fragment method
       module ObjectPatch
-        def cache_fragment(options = {}, &block)
-          fragment ||= Fragment.new(context, options)
-          read_cached_value(fragment) || eval_end_store(fragment, &block)
+        def cache_fragment(object_to_cache = nil, **options, &block)
+          if object_to_cache && block
+            raise ArgumentError, "both object and block could not be passed to cache_fragment"
```

Comment:
We can pass both, for example:

```ruby
def best_comment
  cache_fragment(object) { object.best_comment }
end
```

This is not an object to cache, but an object to generate a cache key. And for convenience we use it as a result if block is not provided to avoid this: `cache_fragment(object) { object }`.

Reply:
Removed the check and added TODO instead of spec to add it later